tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs

```diff
@@ -114,6 +114,73 @@ public void Merge_should_add_expected_stage()
             stages[0].Should().Be("{ $merge : { into : { db : 'database', coll : 'collection' } } }");
         }
 
+        [Fact]
+        public void UnionWith_should_add_expected_stage()
+        {
+            var pipeline = new EmptyPipelineDefinition<BsonDocument>();
+            var withCollection = Mock.Of<IMongoCollection<BsonDocument>>(
+                coll => coll.CollectionNamespace == CollectionNamespace.FromFullName("db.test"));
```

Comment:
I found this line confusing, so I thought I'd share what I think is going on for other reviewers:
We're creating a Mock of `IMongoCollection<BsonDocument>` such that when you call `coll.Collection.NameSpace`, `CollectionNamespace.FromFullName("db.test")` is returned.

The way I'm conceptualizing this is like so:  MOQ can use LINQ expressions to setup the behavior of the mocked interface: however, since LINQ doesn't have an `=` operator, I believe that the `==` operator was probably the only reasonable alternative.

Reply:
Thanks Vincent. I used you as a test subject to see if you could figure out what this line was doing. You were much better at it than me... I had to go back and forth with Dmitry before I finally understood the hidden magic.

I acknowledge that this trick results in more compact code. Not sure that justifies it though.

For now I am OK with leaving this line as is.